polars/polars-io/src/csv/write_impl.rs

```diff
@@ -293,11 +323,18 @@ pub(crate) fn write<W: Write>(
                 for (i, col) in &mut col_iters.iter_mut().enumerate() {
                     let datetime_format = match &options.datetime_format {
                         Some(datetime_format) => Some(datetime_format.as_str()),
```

Comment:
We could prepare this one outside of the while loop as well. Prepare a `Vec` with either `formats` or `datetime_format.as_str` for every column.